CHANGES.rst

```diff
@@ -24,6 +24,13 @@ Axes
 - :py:meth:`AxesManager.set_signal_dimension` is removed, use :py:meth:`~.signal.BaseSignal.as_signal1D`,
   :py:meth:`~.signal.BaseSignal.as_signal2D` or :py:meth:`~.signal.BaseSignal.transpose` of the signal instance instead.
 
+Components
+----------
+
+- The :py:class:`hyperspy._components.polynomial_deprecated.Polynomial` component is removed.
```

Comment:
From the user perspective this may be misleading: it can be read as the `Polynomial` component has been removed, it is its API, which has changed. 

Something along the line of:
```suggestion
- The API of the :py:class:`~.api.model.components1D.Polynomial` has changed after it has been deprecated since hyperspy 1.5. The old API had a single parameters ``coefficients`` and it has been replace with ``a0``, ``a1``, etc.
```